scheduler/src/cook/mesos/api.clj

```diff
@@ -417,12 +418,21 @@
          (s/optional-key :completed) s/Int})
       prepare-schema-response))
 
-(def RawSchedulerRequest
-  "Schema for a request to the raw scheduler endpoint."
+(def JobSubmission
+  "Schema for a request to submit one or more jobs."
   {:jobs [JobRequest]
    (s/optional-key :override-group-immutability) s/Bool
    (s/optional-key :groups) [Group]})
 
+(def RawSchedulerRequestDeprecated
+  "Schema for a request to the raw scheduler endpoint."
+  JobSubmission)
+
+(def JobSubmissionRequest
+  "Schema for a POST request to the /jobs endpoint."
+  (assoc JobSubmission
+    (s/optional-key :pool) s/Str))
```

Comment:
I know we talked about this earlier, but now I'm getting cold feet on having this at the top level. It feels weird when you look at `make-job-txn` for instance, which now has an additional argument where it didn't before. How do you feel about moving this down into each job spec?

Reply:
Some questions come up if move the pool into the job spec:

- should we allow jobs for multiple pools to be submitted together?
- two jobs in the same group probably can't be in different pools?
- if job A cannot be submitted on pool X due to the quota check, should we still allow job B from the same `POST` to be submitted pool Y?

Maybe it would be simpler to just not allow different pools in the same submission. But then, what's the point of requiring users to specify the pool on each job? 

Would it make more sense to put the pool in the URL, like `curl -XPOST localhost:12321/jobs/pool-name -d ...`, instead of having it in the query string?

Reply:
@pschorf Even if we do move the pool into the job spec, I think we will want to keep the extra `pool` argument in `make-job-txn`. The reason is that by the time we get to `make-job-txn`, we're already supposed to have validated the job, which means we've checked that its pool name is valid and the corresponding pool is accepting submissions. We will have resolved the provided name to a pool entity, which we want to pass directly to `make-job-txn` instead of forcing it to re-load the entity by name. It feels odd to add the pool entity to the `Job` schema (especially because we would need to `dissoc` it from the schema for the response side), so I think it actually makes sense to keep the `pool` argument separate.

Overall, I'm not convinced it's a bad idea to have the pool at the top level. Conceptually, you can defend it by saying that you are submitting a set of jobs and groups to a pool.

What do you think?

Reply:
@pschorf Thoughts on this?